astrocut/footprint_cutouts.py

```diff
@@ -0,0 +1,311 @@
+    if verbose:
+        print('Cutout size: ', cutout_size)
+
+    # Get FFI footprints from the cloud
+    s3_uri = 's3://tesscut-ops-footprints/tess_ffi_footprint_cache.json' if product == 'SPOC' \
+        else 's3://tesscut-ops-footprints/tica_ffi_footprint_cache.json'
+    all_ffis = _get_s3_ffis(s3_uri=s3_uri, as_table=True, load_polys=True)
+    if verbose:
+        print(f'Found {len(all_ffis)} footprint files.')
+
+    # Filter FFIs by provided sectors
+    if sequence:
+        # Convert to list
+        if isinstance(sequence, int):
+            sequence = [sequence]
+        all_ffis = all_ffis[np.isin(all_ffis['sequence_number'], sequence)]
+
+        if len(all_ffis) == 0:
+            raise InvalidQueryError('No FFI cube files were found for sequences: ' +
+                                    ', '.join(str(s) for s in sequence))
+        
+        if verbose:
+            print(f'Filtered to {len(all_ffis)} footprints for sequences: {", ".join(str(s) for s in sequence)}')
+
+    # Get sector names and cube files that contain the cutout
+    cone_results = _ra_dec_crossmatch(all_ffis, coordinates, cutout_size, TESS_ARCSEC_PER_PX)
+    if not cone_results:
+        raise InvalidQueryError('The given coordinates were not found within the specified sequence(s).')
+    seq_list = _create_sequence_list(cone_results, product)
+    cube_files_mapping = _get_cube_files_from_sequence_obs(seq_list)
+    if verbose:
+        print(f'Found {len(cube_files_mapping)} matching cube files.')
+    base_file_path = "s3://stpubdata/tess/public/mast/" if product == 'SPOC' \
+        else "s3://stpubdata/tess/public/mast/tica/"
+
+    # Make sure the output directory exists
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    # Executor function to generate cutouts from a cube file
+    def process_file(file):
+        try:
+            factory = CutoutFactory()
+            file_path = os.path.join(base_file_path, file['cube'])
+            output_path = os.path.join(output_dir, file['sectorName'])
+            cutout = factory.cube_cut(
+                file_path,
+                coordinates,
+                cutout_size=cutout_size,
+                product=product,
+                output_path=output_path,
+                threads=8,
+                verbose=verbose
+            )
+            return cutout
+        except Exception as e:
+            warnings.warn(f'Unable to generate cutout from {file_path}: {e}', AstropyWarning)
+            return None
+        
+    # Generate cutout from each cube file
+    cutout_files = []
+    if threads == 'auto' or threads > 1:
+        # Parallel processing
+        if verbose:
+            print(f'Generating cutouts in parallel with max threads: {threads}')
+
+        # Use ThreadPoolExecutor for parallelization
+        max_workers = None if threads == "auto" else threads
+        with ThreadPoolExecutor(max_workers=max_workers) as executor:
```

Comment:
trying this locally I'm not seeing a significant difference between "auto" and `0`. in fact, `0` is sometimes faster, though it is quite variable. I suggest removing this for now, as it adds additional complexity

This is what I was using for benchmarking:

```py
import time
from tempfile import TemporaryDirectory

from astrocut import cube_cut_from_footprint

coordinates = "217.42893801 -62.67949189"
with TemporaryDirectory() as tmp:
    start = time.monotonic()
    vals = cube_cut_from_footprint(coordinates, cutout_size=10, sequence=None, output_dir=tmp, threads=0)
    print("took:", time.monotonic() - start)
```